### PR TITLE
Draft: Add bowtie generator

### PR DESCRIPTION
**Pull request by MartijnVerkleij**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -37,18 +40,81 @@ def create_quantitative_bowtie(training_set_event_tree, training_set_fault_tree,
...
+    # And now we need to union the graph
+    bowtie = Graph(directed=True)
+
+    bowtie._v += fault_tree.vertices + event_tree.vertices
+    bowtie._e += fault_tree.edges + event_tree.edges
+
+    top_events = []
+    # Find the two top events:
+    for vtx in bowtie.vertices:
+        if vtx.label == "TE":
```
Comment (Vescatur) [inline main.py]: hardcoded variable

File: main.py [hunk 2]
```diff
@@ -37,18 +40,81 @@ def create_quantitative_bowtie(training_set_event_tree, training_set_fault_tree,
...
+            top_events.append(vtx)
+
+    assert len(top_events) == 2
+
+    for neighbor in top_events[1].neighbours:
+        bowtie.add_edge(Edge(tail=top_events[0], head=neighbor))
+
+        # Delete the now duplicate top event
+        del top_events[1]
```
Comment (Vescatur) [inline main.py]: del_vertex

File: main.py [hunk 3]
```diff
@@ -5,20 +5,28 @@
...
 
 # noinspection PyTypeChecker
-VertexList = list(Vertex)
+VertexList = [Vertex]
 
+TOP_EVENT_LABEL = "TE"
 
 def start():
     test_bowtie = create_test_bowtie()
-    training_set_event_tree = create_trainingset_event_tree(test_bowtie)
```
Comment (Vescatur) [inline main.py:20]: Reverse this.  Wouter wanted this changed

Comment (wouwouwou) [reply main.py:20]: Did I? 🤔

**Review by wouwouwou — APPROVED** [review]

**Pull request merged**